### PR TITLE
Clarify that the type should be size_t for numClasses in CVBase

### PR DESCRIPTION
**Pull request by rcurtin**
This is a follow-up to #2614.  It just means that when the user tries to construct a `KFoldCV` or `SimpleCV` object, if they pass the number of classes incorrectly (i.e. as an `int`), then the warning will now point out that the type should be `size_t`.

So, for instance, this code does not work:

```
int numClasses = 2;
double validationSize = 0.2;
SimpleCV<DecisionTree<>, Accuracy> scv(validationSize, testData, testLabel, numClasses);
```

but this code does:

```
size_t numClasses = 2;
double validationSize = 0.2;
SimpleCV<DecisionTree<>, Accuracy> scv(validationSize, testData, testLabel, numClasses);
```

So now, if you write the first thing, the error that you get points out that the type of `numClasses` should be `size_t`.

@GLmontanari let me know if you think this message would have saved you time! :+1:

@micyril if you like, feel free to take a look and let me know if you think the error message makes sense or it could be further improved.  I thought it could be cool to adapt the code so that `MetaInfoExtractor` will take an `int` for `numClasses`, but I didn't have the time to dig in.

### CODE REVIEW

**Comment by GLmontanari** [comment]
I think I first made the mistake because I am able to do other actions, I can train the Decision Tree for instance, using `int numClasses` and took for granted that I could to the same also elsewhere.

I agree with you that it would be nice to let `SimpleCV` accept also `int` s.

**Review by zoq — COMMENTED** [review]
It doesn't really matter, but should we keep the static analysis job happy?

**Comment by rcurtin** [comment]
Oops, forgot to push the commit.

File: src/mlpack/core/cv/cv_base_impl.hpp
Comment (micyril) [inline src/mlpack/core/cv/cv_base_impl.hpp:31]: I think this is fine

**Review by micyril — APPROVED** [review]
LGTM, can be merged after fixing checks

**Review by zoq — APPROVED** [review]
Looks good to me as well.

**Comment by zoq** [comment]
Let's wait for the build results before we merge this in just to see if the merge commit worked.

**Comment by rcurtin** [comment]
Thanks @micyril and @zoq!  Looks like everything is passing now, so I'll go ahead and merge. :+1:

**Pull request merged**
